tools/slicec-cs/src/decoding.rs

```diff
@@ -204,8 +198,8 @@ fn decode_result(result_type_ref: &TypeRef<ResultType>, namespace: &str, encodin
     let success_type = &result_type_ref.success_type;
     let failure_type = &result_type_ref.failure_type;
 
-    let decode_success = decode_result_field(success_type, namespace, encoding);
-    let decode_failure = decode_result_field(failure_type, namespace, encoding);
+    let decode_success = decode_type_with_cast(success_type, namespace, encoding, success_type.is_optional).indent();
```

Comment:
Maybe remove the with cast here, it's only used for sequences and dictionaries. What was wrong with the previous name?

Reply:
The previous name was specific to results. Now this logic is re-used for both dictionaries and results.

This function is identical to `decode_func`, it just adds a cast out-front.
So I can't think of a better name than `decode_func_with_cast`.

But I expanded the doc comment explaining when and why to call this function.